test/Microsoft.DotNet.ProjectModel.Tests/GivenThatIWantToLoadAProjectJsonFile.cs

```diff
@@ -0,0 +1,866 @@
+
+        [Fact]
+        public void It_does_not_add_scripts_when_scripts_is_not_a_JsonObject()
+        {
+            var json = new JObject();
+            json.Add("scripts", true);
+            var project = GetProject(json);
+
+            project.Scripts.Should().BeEmpty();
+        }
+
+        [Fact]
+        public void It_does_not_add_the_scripts_when_its_value_is_neither_a_string_nor_an_array_of_strings()
+        {
+            var scriptArrayValues = new [] {"scriptValue2", "scriptValue3"};
+
+            var json = new JObject();
+            var scripts = new JObject();
+            json.Add("scripts", scripts);
+
+            scripts.Add("scriptKey1", "scriptValue1");
+            scripts.Add("scriptKey3", new JArray(scriptArrayValues));
+
+            var project = GetProject(json);
+
+            project.Scripts.Count.Should().Be(2);
+            project.Scripts.First().Key.Should().Be("scriptKey1");
+            project.Scripts.First().Value.Should().Contain("scriptValue1");
+            project.Scripts["scriptKey3"].Should().Contain(scriptArrayValues);
+        }
+
+        [Fact]
+        public void It_throws_when_the_value_of_a_script_is_neither_a_string_nor_array_of_strings()
+        {
+            var json = new JObject();
+            var scripts = new JObject();
+            json.Add("scripts", scripts);
+
+            scripts.Add("scriptKey2", true);
+
+            Action action = () => GetProject(json);
+
+            action.ShouldThrow<FileFormatException>()
+                .WithMessage("The value of a script in project.json can only be a string or an array of strings");
+        }
+
+        [Fact]
+        public void It_uses_an_empty_compiler_options_when_one_is_not_set_in_the_ProjectJson()
+        {
+            _emptyProject.GetCompilerOptions(null, null).Should().Be(new CommonCompilerOptions
+            {
+                OutputName = ProjectName
+            });
+        }
+
+        [Fact]
+        public void It_sets_analyzerOptions_when_it_is_set_in_the_compilationOptions_in_the_ProjectJson()
+        {
+            var json = new JObject();
+            var compilationOptions = new JObject();
+            json.Add("compilationOptions", compilationOptions);
+
+            var analyzerOptions = new JObject();
+            compilationOptions.Add("analyzerOptions", analyzerOptions);
+
+            analyzerOptions.Add("languageId", "C#");
+
+            var project = GetProject(json);
+            project.AnalyzerOptions.LanguageId.Should().Be("C#");
+        }
+
+        [Fact]
+        public void It_throws_when_the_analyzerOptions_languageId_is_no_a_string()
+        {
+            var json = new JObject();
+            var compilationOptions = new JObject();
+            json.Add("compilationOptions", compilationOptions);
+
+            var analyzerOptions = new JObject();
+            compilationOptions.Add("analyzerOptions", analyzerOptions);
+
+            analyzerOptions.Add("languageId", true);
```

Comment:
We need a better model here, eventually. We only prove that booleans don't work at the moment...